src/dmd/backend/dwarfdbginf.d

```diff
@@ -2844,14 +2843,12 @@ static if (1)
             case TYwchar_t:
             {
                 uint code3 = dwarf_abbrev_code(abbrevWchar.ptr, (abbrevWchar).sizeof);
-                uint typebase = dwarf_typidx(tstypes[TYint]);
                 idx = cast(uint)debug_info.buf.length();
+
                 debug_info.buf.writeuLEB128(code3);       // abbreviation code
                 debug_info.buf.writeString("wchar_t");    // DW_AT_name
-                debug_info.buf.write32(typebase);         // DW_AT_type
-                debug_info.buf.writeByte(1);              // DW_AT_decl_file
-                debug_info.buf.write16(1);                // DW_AT_decl_line
-                typidx_tab[ty] = idx;
+                debug_info.buf.writeByte(tysize(TYint));  // DW_AT_byte_size
+                debug_info.buf.writeByte(DW_ATE_signed);  // DW_AT_encoding
```

Comment:
this is technically signed because it mimics the `int` type. This behaviour complies with clang/gcc debug info.